psydac/linalg/block.py

```diff
@@ -782,47 +767,53 @@ def __rmul__(self, a):
 
     # ...
     def __add__(self, M):
-        assert isinstance(M, BlockMatrix)
-        assert M.  domain is self.  domain
-        assert M.codomain is self.codomain
-        blocks  = {}
-        for ij in set(self._blocks.keys()) | set(M._blocks.keys()):
-            Bij = self[ij]
-            Mij = M[ij]
-            if   Bij is None: blocks[ij] = Mij.copy()
-            elif Mij is None: blocks[ij] = Bij.copy()
-            else            : blocks[ij] = Bij + Mij
-        mat = BlockMatrix(self.domain, self.codomain, blocks=blocks)
-        if len(mat._blocks) != len(self._blocks):
-            mat.set_backend(self._backend)
-        elif self._backend is not None:
-            mat._func = self._func
-            mat._args = self._args
-            mat._blocks_as_args = [mat._blocks[key]._data for key in self._blocks]
-            mat._backend = self._backend
-        return mat
+        if isinstance(M, BlockLinearOperator):
+            #assert isinstance(M, BlockLinearOperator)
+            assert M.  domain is self.domain
+            assert M.codomain is self.codomain
+            blocks  = {}
+            for ij in set(self._blocks.keys()) | set(M._blocks.keys()):
+                Bij = self[ij]
+                Mij = M[ij]
+                if   Bij is None: blocks[ij] = Mij.copy()
+                elif Mij is None: blocks[ij] = Bij.copy()
+                else            : blocks[ij] = Bij + Mij
+            mat = BlockLinearOperator(self.domain, self.codomain, blocks=blocks)
+            if len(mat._blocks) != len(self._blocks):
+                mat.set_backend(self._backend)
+            elif self._backend is not None:
+                mat._func = self._func
+                mat._args = self._args
+                mat._blocks_as_args = [mat._blocks[key]._data for key in self._blocks]
+                mat._backend = self._backend
+            return mat
+        else:
+            return LinearOperator.__add__(self, M)
```

Comment:
In order to minimize the number of lines changed in this file, I suggest to revert the body of this function back to its original content, and simply add two lines at the beginning of it:
```python
def __add__(self, M):
    if not isinstance(M, BlockLinearOperator):
        return LinearOperator.__add__(self, M)

    # Original code here
```
The same comment applies to the functions `__sub__`, `__iadd__` and `__isub__` below.